### ISSUE TITLE
Old test results should have an ability to be discarded/deleted

### ISSUE THREAD

**Issue by Sedemicra** [GHPReporter/Ghpr.Core#57]
I know it is possible to determine how many runs are **displayed** in the report html. Is it also possible to determine how many is kept in general? 
I will be having many runs and over time the logs will grow massive, which I would like to avoid. I would like to be able to set a limit based on either a number of runs or time to completely delete all older logs. I haven't been able to find such possibility. Is there one? If not, would it be possible to have such feature implemented?
Thanks in advance!

**Comment by elv1s42** [comment]
Hi @axinom-raido !

You can set up the default amount of test runs and tests to be loaded by editing settings `.json` file:
https://github.com/GHPReporter/Ghpr.Core#about-settings-file

Deleting old test runs may be implemented later as part of Ghpr.Console functionality.

Thank you

**Comment by axinom-raido** [comment]
Thank you for confirming my suspicions. 
Being able to set the limit of how many testruns are preserved in total would be much appreciated feature as it would help to keep total logs in reasonable size without losing the latest results. I guess I need to look for an alternative until such option is available. 
Thanks again for quick answer.

**Comment by elv1s42** [comment]
Hi @axinom-raido !
Can you please let me know what exact data you want to able to delete?

The plan is to implement this feature as part of Ghpr.Console application and give users an ability to detele old runs (probably by passing date argument into console app and deleting all data older than the given date)

**Comment by elv1s42** [comment]
Hi @axinom-raido !
Thank you very much for detailed explanation. Agreed to create "clean up" job that can be run after a new report is generated.

**Comment by elv1s42** [comment]
Now old runs can be deleted using `retention` section from reporter `.json` settings: 
 ``` json
  "retention": {
    "amount": 10,
    "till": "2018-06-29 10:00:00"
  }
 ```

This clean up job will run each time after the new report is generated. 
`amount` - total runs that will be left, all other will be deleted. 
`till` - all runs with finish date older than  this value will be deleted too. 

There is no way to restore the data, so please use these settings carefully.

**Comment by elv1s42** [comment]
Released in v0.9.2

**Comment by Sedemicra** [comment]
Thank you. Will try this out once I get time.
